### DISCUSSION TITLE
Fluence in Spectral Distributions

### DISCUSSION THREAD

**Discussion by LEley22** [nrc-cnrc/EGSnrc#1045] [Physics] [answered]
Hi!
I've been using beamdp to plot the phase space output from my BEAMnrc simulation, and I'm interested in the energy spectrum so I've been plotting the spectral distributions. I am confused about the calculation of the fluence on the y-axis - is this normalised by the incident beam energy as well as the particle number?
If anybody knows the specific calculation of this please let me know - I find the definition to in the beamdp User Manual to be equally confusing as it discusses the weighting, but my particles are all equally weighted.

Additionally, I am confused by their definition of planar vs estimated real fluence so if anybody has a better understanding of this quantity and can give me an alternative explanation, I'd hugely appreciate it!

Thank you,
Lauryn

**Comment by rtownson** [comment]
For estimated real fluence, particles on an angle count less than particles aimed "straight down". If they are straight down and all weights are 1, then they count as 1. If they are on an angle, they count as `value=1/(MAX(0.08716,ABS(WIN))`, where `ABS` is the absolute value, `MAX` chooses the maximum between two values, and `WIN` is the z direction cosine.

For energy fluence, the `value` is also multiplied by the energy of the particle. For both regular fluence and energy fluence, it is normalized by the number of primary histories.

**Reply by ftessier** [reply, marked as answer]
Note that the constant `0.08716` here is to prevent a numerical divergence of $\large 1/\\!\cos \theta$, when $\large \cos\theta \rightarrow 0$ (particle nearly parallel to the surface). 

- The so-called planar fluence is simply the number of particles divided by the surface area regardless of their directions.
 
- The estimated real fluence includes a $\large 1/\\!\cos \theta$ correction factor to take into account that a particles that crosses the surface at an oblique angle contributes to fluence at "more points" on the surface, very loosely speaking. 

In more formal terms: the fluence contribution of a particle over a volume is the path length inside the volume, divided by that volume; this is a direct consequence of the definition of the fluence. If you think of a differential thickness around the surface in question (creating a thin volume around the surface), a particle crossing the surface at an oblique angle leads to a longer track inside the differential volume, and this length increases as $\large 1/\\!\cos \theta$, as can be seen from a geometrical sketch.

Another way to understand the $\large 1/\\!\cos \theta$ factor is to realize that fluence is a _non-directional quantity:_ it is a scalar which does _not_ depend on the direction of the particles. Imagine then a uniform parallel beam crossing some plane with normal incidence. Now tilt the plane: the number of particles crossing the plane per unit area—the _planar_ fluence—decreases on account of the angle. However the _real_ fluence must remain the same, since it does not depend on the particle direction. Therefore a correction is required, and this correction is, again, $\large 1/\\!\cos \theta$.

**Reply by LEley22** [reply]
Thank you for your help, that helps a lot! 

To make this question more specific - I am using BEAMnrc to compare to an x-ray spectrum I have generated in Geant4 by recreating the exact geometry. The x-ray spectrum outputted by BEAMnrc looks identical in trend to that of Geant4, but the y-axis number seems to be out by a factor of 3. I am comparing planar fluence in the spectral distribution and normalising the Geant4 data to match this by dividing the y-axis counts by incident particle number and detector area, based on your replies above. Is there something obviously wrong in this method causing the factor-3 discrepancy?

**Reply by ftessier** [reply]
There is nothing wrong with that, provided you are considering the spectrum as a probability distribution, for example to sample photon energies for a subsequent simulation or calculation; then the distribution has to be normalized before sampling anyways. Ideally though, you would want to understand the origin of the discrepancy.

If you are using the spectrum in absolute terms, for example to calculate the actual number of photons of a specified energy _per electron incident on the x-ray target,_ then the simulations ought to **give the same numerical results,** because the number of incident electrons is known exactly.  Double-check the normalization in `beamdp`, it often plots results _per incident fluence_ or other normalization.

This is sometimes misunderstood, because spectra are typically compared on relative terms, especially when compared to experiments, where the number of incident electrons is not known precisely (beyond the tube current).
